### PR TITLE
[AudioBoom] Added extractor for AudioBoom

### PR DESCRIPTION
**Pull request by benjamincongdon**
Added simple extractor for AudioBoom.com as requested in #8796.

### CODE REVIEW

File: youtube_dl/extractor/audioboom.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class AudioBoomIE(InfoExtractor):
+    _VALID_URL = r'https?:\/\/(?:www\.)?audioboom\.com\/boos\/(?P<id>[0-9]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/audioboom.py]: No need to escape forward slash.

**Comment by benjamincongdon** [comment]
Fixed unnecessarily escaped forward slashes.

File: youtube_dl/extractor/audioboom.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            'title': 'AI Safety and The Legacy of Bletchley Park',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._og_search_title(webpage)
+        _id = self._search_regex(r'https?://(?:www\.)?audioboom\.com/boos/(?P<id>[0-9]+)', url, 'id')
```
Comment (dstftw) [inline youtube_dl/extractor/audioboom.py]: This matches just any plain text that meet this pattern. Instead it should extract it via pattern that has embed semantics. This also won't work if actual media is stored under different URI.

File: youtube_dl/extractor/audioboom.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+        download_url = self._search_regex(r'^(?P<url>.*?)(\?|$)', url, 'url') + '.mp3'
```
Comment (dstftw) [inline youtube_dl/extractor/audioboom.py]: What if it's not an mp3?

**Comment by dstftw** [comment]
Description and duration should also be extracted.

**Comment by benjamincongdon** [comment]
Description and duration now extracted. Now get download URL from `<meta>` tags.

File: youtube_dl/extractor/audioboom.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._og_search_title(webpage)
+
+        download_url = self._search_regex(
+            r'<meta content=\'(?P<url>[A-Za-z0-9.\-\:/]+)\' property=\'og:audio\'>', webpage, 'url')
```
Comment (dstftw) [inline youtube_dl/extractor/audioboom.py]: Use `self._og_search_property` instead.

File: youtube_dl/extractor/audioboom.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+        duration = float_or_none(self._search_regex(
+            r'<meta content=\'(?P<duration>[0-9.]+)\' property=\'weibo:audio:duration\'>', webpage, 'duration'))
```
Comment (dstftw) [inline youtube_dl/extractor/audioboom.py]: Use `self._og_search_property` instead. Duration extraction must not be `fatal`.

**Comment by benjamincongdon** [comment]
Duration extraction now not `fatal`. URL extracted now using `_og_search_property()`

**Comment by dstftw** [comment]
Cherry-picked in 61f317c24c040d051ec6652cb0d4c650c1fc0361, thanks.

**Pull request closed without merging**